src/lib/transaction_logic/parties_logic.ml

```diff
@@ -1405,22 +1409,43 @@ module Make (Inputs : Inputs_intf) = struct
     (* DO NOT ADD ANY UPDATES HERE. They must be earlier in the code.
        See comment above.
     *)
+    let implicit_account_creation_fee =
+      Party.implicit_account_creation_fee party
+    in
+    (* Check the token for implicit account creation fee payment. *)
+    let local_state =
+      Local_state.add_check local_state Cannot_pay_creation_fee_in_token
+        Bool.((not implicit_account_creation_fee) ||| party_token_is_default)
+    in
+    (* Compute the change to the account balance. *)
+    let local_state, local_delta =
+      let local_delta = Amount.Signed.negate (Party.balance_change party) in
+      let local_delta_for_creation, `Overflow creation_overflow =
+        let open Amount.Signed in
+        add_flagged local_delta (of_unsigned account_creation_fee)
+      in
+      let pay_creation_fee =
+        Bool.(account_is_new &&& implicit_account_creation_fee)
+      in
+      let creation_overflow = Bool.(pay_creation_fee &&& creation_overflow) in
+      let local_delta =
+        Amount.Signed.if_ pay_creation_fee ~then_:local_delta_for_creation
```

Comment:
In the case of default-token accounts, are we deducting account creation fee for an account twice? 
And, for token accounts what's the rationale for burning custom tokens?